src/components/Alphabet/AlphabetData.js

```diff
@@ -1,74 +1,55 @@
 import { useEffect, useState } from 'react'
-import { useNavigate, useParams } from 'react-router-dom'
-import { useQuery } from 'react-query'
+import { useParams, useSearchParams } from 'react-router-dom'
 
 // FPCC
-import { useSiteStore } from 'context/SiteContext'
-import api from 'services/api'
+import useAlphabet from 'common/dataHooks/useAlphabet'
 
-const AlphabetData = ({ widgetView }) => {
-  const { site } = useSiteStore()
-  const { uid } = site
+const AlphabetData = () => {
   const { sitename } = useParams()
   const [selectedData, setSelectedData] = useState({})
-  const navigate = useNavigate()
+  const [searchParams] = useSearchParams()
 
-  const character = new URLSearchParams(document.location.search).get('char')
-    ? new URLSearchParams(document.location.search).get('char')
-    : null
+  const character = searchParams.get('char') || null
 
-  const { status, isLoading, error, isError, data } = useQuery(
-    ['alphabet', uid],
-    () => api.alphabet.get(uid),
-    {
-      enabled: !!uid,
-      refetchOnWindowFocus: false,
-      refetchOnReconnect: false,
-    },
-  )
+  const { status, isLoading, data } = useAlphabet()
 
   // Find slected character data
-  const findSelectedCharacterData = (selectedCharacter) => {
+  const findCharacterData = (selectedCharacter) => {
     const characters = Object.assign([], data?.characters)
     const found = characters.filter(
       (char) => char.title === selectedCharacter,
     )[0]
     return found
   }
 
-  useEffect(() => {
-    if (character && data?.characters?.length > 0) {
-      const _selectedData = findSelectedCharacterData(character)
-      if (_selectedData && _selectedData?.title !== selectedData?.title) {
-        setSelectedData(_selectedData)
-      }
-    }
-    if (!character && data?.characters?.length > 0) {
-      const _selectedData = findSelectedCharacterData(
-        data?.characters?.[0]?.title,
-      )
-      if (_selectedData && _selectedData?.title !== selectedData?.title) {
-        setSelectedData(_selectedData)
-      }
+  const setCharacterDataToDisplay = (characterTitle) => {
+    const dataForCharacter = findCharacterData(characterTitle)
+    if (dataForCharacter && dataForCharacter?.title !== selectedData?.title) {
+      setSelectedData(dataForCharacter)
     }
-  }, [character, data])
+  }
 
   useEffect(() => {
-    if (isError && !widgetView) {
-      navigate(
-        `/${sitename}/error?status=${error?.response?.status}&statusText=${error?.response?.statusText}&url=${error?.response?.url}`,
-        { replace: true },
```

Comment:
Is the error display handled another way now?

Reply:
Discussed 👍